extensions/civetweb/tests/ListenHTTPTests.cpp

```diff
@@ -0,0 +1,514 @@
+
+    // Create temporary directories
+    testController.createTempDirectory(tmp_dir);
+    REQUIRE(tmp_dir != nullptr);
+
+    // Define test input file
+    std::string test_input_file = utils::file::FileUtils::concat_path(tmp_dir, "test");
+    {
+      std::ofstream os(test_input_file);
+      os << "Hello response body" << std::endl;
+    }
+
+    // Build MiNiFi processing graph
+    plan = testController.createPlan();
+    get_file = plan->addProcessor(
+        "GetFile",
+        "GetFile");
+    update_attribute = plan->addProcessor(
+        "UpdateAttribute",
+        "UpdateAttribute",
+        core::Relationship("success", "description"),
+        true);
+    listen_http = plan->addProcessor(
+        "ListenHTTP",
+        "ListenHTTP",
+        core::Relationship("success", "description"),
+        true);
+    log_attribute = plan->addProcessor(
+        "LogAttribute",
+        "LogAttribute",
+        core::Relationship("success", "description"),
+        true);
+
+    // Configure GetFile processor
+    plan->setProperty(get_file, "Input Directory", tmp_dir);
+
+    // Configure UpdateAttribute processor
+    plan->setProperty(update_attribute, "http.type", "response_body", true);
+
+    // Configure ListenHTTP processor
+    plan->setProperty(listen_http, "Listening Port", "0");
+  }
+
+  virtual ~ListenHTTPTestsFixture() {
+    free(tmp_dir);
+    LogTestController::getInstance().reset();
+  }
+
+  void create_ssl_context_service(const char* ca, const char* client_cert) {
+    auto config = std::make_shared<minifi::Configure>();
+    if (ca != nullptr) {
+      config->set(minifi::Configure::nifi_security_client_ca_certificate, utils::file::FileUtils::get_executable_dir() + "/resources/" + ca);
+    }
+    if (client_cert != nullptr) {
+      config->set(minifi::Configure::nifi_security_client_certificate, utils::file::FileUtils::get_executable_dir() + "/resources/" + client_cert);
+      config->set(minifi::Configure::nifi_security_client_private_key, utils::file::FileUtils::get_executable_dir() + "/resources/" + client_cert);
+      config->set(minifi::Configure::nifi_security_client_pass_phrase, "Password12");
+    }
+    ssl_context_service = std::make_shared<minifi::controllers::SSLContextService>("SSLContextService", config);
+    ssl_context_service->onEnable();
+  }
+
+  void run_server() {
+    plan->runNextProcessor(); // GetFile
+    plan->runNextProcessor(); // UpdateAttribute
+    plan->runNextProcessor(); // ListenHTTP
+
+    auto raw_ptr = dynamic_cast<org::apache::nifi::minifi::processors::ListenHTTP*>(listen_http.get());
+    std::string protocol = std::string("http") + (raw_ptr->isSecure() ? "s" : "");
+    std::string portstr = raw_ptr->getPort();
+    REQUIRE(LogTestController::getInstance().contains("Listening on port " + portstr));
+
+    url = protocol + "://localhost:" + portstr + "/contentListener/" + endpoint;
+  }
+
+  void test_connect(bool should_succeed = true, int64_t response_code = 200) {
```

Comment:
The naming might be misleading, but should_succeed means that we get an HTTP response (and not for example, die in the TLS connection phase because of a bad cert) and the response_code means the error code we get. Because our processor specification includes specific error codes for specific cases (403 on AuthorizedDNPattern mismatch, 405 on unsupported HTTP methods), I want to specifically test for those.

Reply:
Okay, makes sense, thanks!